### PR TITLE
MULE-8569: Applications and corresponding domains should share the same ...

### PR DESCRIPTION
**Pull request by marianogonzalez**
...OptionalObjectController

### CODE REVIEW

File: modules/spring-config/src/main/java/org/mule/config/spring/CompositeOptionalObjectsController.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        for (OptionalObjectsController controller : controllers)
+        {
+            controller.discardOptionalObject(key);
+        }
+    }
+
+    @Override
+    public boolean isOptional(String key)
+    {
+        for (OptionalObjectsController controller : controllers)
```
Comment (pablokraan) [inline modules/spring-config/src/main/java/org/mule/config/spring/CompositeOptionalObjectsController.java]: Should not be optional when is optional in all the controllers?

File: modules/spring-config/src/main/java/org/mule/config/spring/CompositeOptionalObjectsController.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            }
+        }
+
+        return false;
+    }
+
+    @Override
+    public boolean isDiscarded(String key)
+    {
+        for (OptionalObjectsController controller : controllers)
```
Comment (pablokraan) [inline modules/spring-config/src/main/java/org/mule/config/spring/CompositeOptionalObjectsController.java]: idem

File: modules/spring-config/src/main/java/org/mule/config/spring/CompositeOptionalObjectsController.java (hunk 1 above)
Comment (marianogonzalez) [inline modules/spring-config/src/main/java/org/mule/config/spring/CompositeOptionalObjectsController.java]: no, it has to be on at least one. If the optional entry is in the domain, then the one from the app will return false but you still have to ignore the exception

File: modules/spring-config/src/main/java/org/mule/config/spring/CompositeOptionalObjectsController.java (hunk 2 above)
Comment (marianogonzalez) [inline modules/spring-config/src/main/java/org/mule/config/spring/CompositeOptionalObjectsController.java]: idem too

File: modules/spring-config/src/main/java/org/mule/config/spring/MuleArtifactContext.java [hunk 3]
```diff
@@ -75,10 +77,45 @@ public MuleArtifactContext(MuleContext muleContext, ConfigResource[] configResou
...
     }
 
+    /**
+     * Parses configuration files creating a spring ApplicationContext which is used
+     * as a parent registry using the SpringRegistry registry implementation to wraps
+     * the spring ApplicationContext
+     *
+     * @param muleContext               the {@link MuleContext} that own this context
+     * @param configResources           the configuration resources to use
+     * @param optionalObjectsController the {@link DefaultOptionalObjectsController} to use. Cannot be {@code null}
```
Comment (pablokraan) [inline modules/spring-config/src/main/java/org/mule/config/spring/MuleArtifactContext.java]: DefaultOptionalObjectsController > OptionalObjectsController

File: modules/spring-config/src/main/java/org/mule/config/spring/MuleArtifactContext.java [hunk 4]
```diff
@@ -75,10 +77,45 @@ public MuleArtifactContext(MuleContext muleContext, ConfigResource[] configResou
...
+    }
+
+    /**
+     * Parses configuration files creating a spring ApplicationContext which is used
+     * as a parent registry using the SpringRegistry registry implementation to wraps
+     * the spring ApplicationContext
+     *
+     * @param muleContext               the {@link MuleContext} that own this context
+     * @param springResources           the configuration resources to use
+     * @param optionalObjectsController the {@link DefaultOptionalObjectsController} to use. Cannot be {@code null}
```
Comment (pablokraan) [inline modules/spring-config/src/main/java/org/mule/config/spring/MuleArtifactContext.java]: idem

**Pull request closed without merging**